lua/wire/stools/expression2.lua

```diff
@@ -55,21 +72,137 @@ if SERVER then
+		BetterChatPrint(initiator, "E2 view request sent for '"..truncName.."' owned by "..chip.player:Nick())
+		BetterChatPrint(chip.player, "You just received a request to view your E2 '"..truncName.."' from "..initiator:Nick()..", which you can view in your context menu")
+
+		-- Add the request data to the local requests table for when the request from the client comes in
+		viewRequests[index] = {
+			initiator = initiator,
+			owner = chip.player,
+			name = truncName,
+			chip = chip,
+			expiry = CurTime() + 60 -- 1 minute for the request before it's invalidated (could make this a convar)
+		}
+
+		net.Start("WireExpression2_ViewRequest")
+			net.WriteInt(chip:EntIndex(), 32) -- For lookup in the request table when the owner answers
+			net.WriteFloat(CurTime() + 60)            -- For making requests expire on time clientside
+			net.WriteEntity(initiator)                -- The player attempting to view the E2
+			net.WriteString(truncName)                -- Name of the E2 so the owner knows what they're agreeing to
+			net.WriteEntity(chip)             -- Chip entity for validation clientside
+		net.Send(chip.player)
 	end
 
+	util.AddNetworkString("WireExpression2_OpenEditor")
+	util.AddNetworkString("WireExpression2_ViewRequest")
+	util.AddNetworkString("WireExpression2_AnswerRequest")
+	function TOOL:Think()
+		--[[
+			I had to replace TOOL:RightClick with TOOL:Think as prop protection was preventing
+			the view requests system from functioning as intended
+
+			So this manually handles right click meaning people don't need to give each other
+			full prop protection permissions in order to share a chip
+		]]
+		if not IsFirstTimePredicted() then return end
+
+		local player = self:GetOwner()
+		if player:KeyPressed(IN_ATTACK2) then
+			local chip = player:GetEyeTrace().Entity
+			if chip:IsPlayer() then return end
+
+			local player = self:GetOwner()
+
+			if IsValid(chip) and chip:GetClass() == "gmod_wire_expression2" then
+				if chip.player == player or bypassList[player:SteamID()] then -- Just download if the toolgun user owns this chip (or they're in the bypass list)
+					self:Download(player, chip)
+					player:SetAnimation(PLAYER_ATTACK1)
+				elseif not IsValid(chip.player) then -- If the chip has no valid owner we can't send a request, so do a CanTool check
+					if hook.Run("CanTool", player, WireLib.dummytrace(chip), "wire_expression2") then
+						self:Download(player, chip)
+						player:SetAnimation(PLAYER_ATTACK1)
+					end
+				else
+					RequestView(chip, player)
+					player:SetAnimation(PLAYER_ATTACK1)
+				end
+			else
+				net.Start("WireExpression2_OpenEditor") net.Send(player)
+			end
+		end
+	end
+	net.Receive("WireExpression2_AnswerRequest", function(len, plr)
+		local accept, index, initiator, name -- note that name, and strictly initiator aren't required to function, however it makes it harder for a client to spoof info
+		local success = pcall(function()
```

Comment:
This isn't going to stop malformed messages. They don't error.

Reply:
Hm, would have expected net functions to error on invalid bytes (like trying to read a string from an empty netmsg).  
I'll just add some manual checks below then, I assume they would just return `nil` if they don't error but fail to read?

Reply:
They won't return nil either. They just return garbage.

Reply:
Wow not only do they not fail, they return valid datatypes from the read

Reply:
Ok the return values actually seem to change sometimes, `net.ReadInt(32)` seems to usually return `0`, yet returned `777` on one request, however even if all parameters somehow become valid, and they match a valid view request, then they'd still need to be sent from the owning player's client, so with some better validation on my part, shouldn't be an issue.